### ISSUE TITLE
Code Completion Finetuning on CodeSearchNet

### ISSUE THREAD

**Issue by wannita901** [parshinsh/rl_code#7]
Hi! Thank you for your amazing work!

I wonder how can I obtain the code completion fine-tuning checkpoint. As to be able to compile the code in Python, the newline and indentation information are required. How do you preprocess the CodeSearchNet dataset? And How do you define the last 25 tokens?

**Comment by parshinsh** [comment]
@wannita901 Thanks for your interest! 
To begin, use the Python snippets from the [CodeSearchNet](https://huggingface.co/datasets/code_search_net) dataset. 
For compilation evaluation, refer to `terminal_compiler.py` and `test.py` in the `./compiler/` folder [here](https://github.com/reddy-lab-code-research/PPOCoder/blob/main/compiler/), which handle newline and indentation. The last 25 tokens are based on tokens from codeT5 vocabulary. 
We haven't released code completion fine-tuning checkpoints yet, but may do so in the future.

**Comment by wannita901** [comment]
Thank you for your response. Just to be a little bit more clear. So the last 25 tokens for GPT-like models (e.g., GPT-2, CodeGPT, etc.) are also based on tokens from codeT5 vocabulary?

**Comment by parshinsh** [comment]
Yes, we use fixed masked tokens across baselines for a fair comparison.

**Comment by wannita901** [comment]
Hi, thank you for the reply! actually, I still have some questions. I see some of your results (Table 1 in your paper) are exactly the same number as [this paper](https://aclanthology.org/2022.findings-acl.2/) (Table 1). But I can't find any dataset or training code from the previous work. May I ask:

1. Do you run their replication or the results are referenced from the previous paper? 
2. If you run the replication, do you use the same dataset sampling as them? (CodeSearchNet has 1.1M Python data, but I see the training in both papers uses only 40-45k.)
3. You said the masked tokens are fixed from codeT5 vocabulary, how did you achieve the same results while the previous work didn't use codeT5?
4. Just to be clear, how do you calculate the exact match? Does it mean all the last 25 predicted token is exactly matched to the last 25 tokens of ground truth? or it is calculated by token-level match?

Thank you so much for the interesting work!
